bin/anvi-analyze-synteny

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python
+# -*- coding: utf-8
+"""anvi-analyze-synteny ... let's extract some order from this chaos!
+
+What do we say to loci that appear to have no coherent synteny patterns...?
+
+... not today ;)
+
+The goal of this program is to analyze the synteny patterns within a group of loci. The main idea is to have a sliding window 
+that will record a sequence of genes of size N to form an Ngram (similar to a kmer profile). This program will output
+a tsv with counts for each Ngram per contig provided. The resulting Ngram count table can then be used 
+to analyze the presence and absence of groups of genes by the user.
+
+To test this code run:
+            
+            >>> cd anvio/anvio/tests
+            >>> ./run_all_tests.sh
+            >>> output_dir=sandbox/test-output
+
+            # make a external-genomesfile
+            >>> echo -e "name\tcontigs_db_path\ng01\t$output_dir/01.db\ng02\t$output_dir/02.db\ng03\t$output_dir/03.db" > $output_dir/external-genomes-file.txt
+
+            Default test that ignores ngrams with unknown functions:
+
+            >>> anvi-analyze-synteny -e $output_dir/external-genomes-file.txt \
+                                     --annotation-source COG_FUNCTION \
+                                     --window-range 2:3 \
+                                     -o $output_dir/synteny_output_no_unknowns.tsv
+
+            >>> anvi-analyze-synteny -e $output_dir/external-genomes-file.txt \
+                                     --annotation-source COG_FUNCTION \
+                                     --window-range 2:3 \
+                                     -o $output_dir/synteny_output_with_unknowns.tsv \
+                                     --analyze-unknown-functions
+
+            Unknown-function-mode test that records ngrams with unknown functions:\
+
+            >>> anvi-analyze-synteny -e $output_dir/external-genomes-cps.txt \
+                                     --annotation-source COG_FUNCTION \
+                                     --window-range 2:3 \
+                                     -o $output_dir/tsv.txt \
+                                     --analyze-unknown-functions
+"""
+
+import sys
+import argparse
+
+import anvio
+import anvio.terminal as terminal
+
+from anvio.synteny import NGram
+from anvio.errors import ConfigError, FilesNPathsError
+
+
+__author__ = "Developers of anvi'o (see AUTHORS.txt)"
+__copyright__ = "Copyleft 2015-2018, the Meren Lab (http://merenlab.org/)"
+__credits__ = []
+__license__ = "GPL 3.0"
+__version__ = anvio.__version__
+__maintainer__ = "Matthew S. Schechter"
+__email__ = "mschechter@uchicago.edu"
+__requires__ = ['contigs-db', 'external-genomes', 'annotation-source', 'ngram-window-range']
```

Comment:
we can't just put any keyword here :) __requires__ and __provides__ must be defined here:

https://github.com/merenlab/anvio/blob/master/anvio/programs.py#L38

Reply:
ahhhh I didn't know that! Ok great Ill fix that

Reply:
fixed!